bson/primitive/primitive.go

```diff
@@ -223,9 +265,201 @@ type E struct {
 //	bson.M{"foo": "bar", "hello": "world", "pi": 3.14159}
 type M map[string]interface{}
 
+// UnmarshalJSON decodes M from JSON.
+func (m *M) UnmarshalJSON(b []byte) error {
```

Comment:
Why do we need to define custom JSON unmarshaling logic for `bson.M`? Other than the type names being different, the JSON unmarshaling logic seems identical.